### PR TITLE
quotemark rule does not apply to JSON files

### PR DESCRIPTION
**Pull request by aprilrd**
#### PR checklist

- [x] Addresses an existing issue: #4089
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:

quotemark rule does not apply to JSON files

#### Is there anything you'd like reviewers to focus on?

I am not sure if using filename is the best way to detect json files.

#### CHANGELOG.md entry:

[bugfix] quotemark rule does not apply to JSON files

### CODE REVIEW

File: src/rules/quotemarkRule.ts [hunk 1]
```diff
@@ -91,6 +93,15 @@ export class Rule extends Lint.Rules.AbstractRule {
         const quoteMark = getQuotemarkPreference(args);
         const jsxQuoteMark = getJSXQuotemarkPreference(args);
 
+        if (sourceFile.fileName.endsWith(".json")) {
+            return this.applyWithFunction(sourceFile, walk, {
+                avoidEscape: true,
+                avoidTemplate: true,
+                jsxQuoteMark: '"',
+                quoteMark: '"',
+            });
```
Comment (JoshuaKGoldberg) [inline src/rules/quotemarkRule.ts]: Shouldn't TypeScript already give errors for invalid JSON? Can the rule skip these files altogether?

I'm also wondering if all rules should skip JSON files, but that's a question for another issue...

Comment (aprilrd) [reply src/rules/quotemarkRule.ts]: Created a separate issue for all rules: https://github.com/palantir/tslint/issues/4426

Comment (JoshuaKGoldberg) [reply src/rules/quotemarkRule.ts]: Thanks!

**Comment by adidahiya** [comment]
I'd rather fix https://github.com/palantir/tslint/issues/4426 (skip all JSON file linting) than implement this band aid. I will try to get that in for v5.12.1.

**Pull request closed without merging**